pkg_resources/__init__.py

```diff
@@ -2477,11 +2477,26 @@ def _dep_map(self):
                             extra, marker = extra.split(':', 1)
                             if invalid_marker(marker):
                                 # XXX warn
-                                reqs=[]
+                                reqs = []
                             elif not evaluate_marker(marker):
-                                reqs=[]
+                                reqs = []
                         extra = safe_extra(extra) or None
-                    dm.setdefault(extra,[]).extend(parse_requirements(reqs))
+                    try:
+                        dm.setdefault(extra, []).extend(
+                            parse_requirements(reqs))
+                    except (
+                        packaging.requirements.InvalidRequirement,
+                        RequirementParseError
+                    ) as e:
+                        raise RequirementParseError(
```

Comment:
I usually try to avoid flattening the details of an exception into a string, but instead supply the relevant parts as arguments to the exception and have the exception render those parts to a string. I'm not familiar with the structure of RequirementParseError, so it may not be possible to do that here if retaining the original exception is important.